### PR TITLE
Read lsync, nrows, nsamp, clockMHz from cringeGlobals.json

### PR DESCRIPTION
**Pull request by ggggggggg**
This changes `LanceroSource.Configure` to read from `~/.cringe/cringeGlobals.json` to learn `lsync`, `nrows`, `Nsamp`, and `clockMHz`. In `sampleCard` where we calculate `nrows` and `lsync` there is now an error if `nrows` doesn't match, and a warning if `lsync` doesn't match. 

Tests that used `LanceroSource.Configure` to set these values now have to reach inside of `LanceroSource` to set these values directly, so they are a bit uglier. Also for testing, it reads from `lancero/test_data/cringeGlobals.json` since `~/.cringe/cringeGlobals.json` won't exist on travis or my laptop.

I think this will work without any modifications to dastard commander, since Dastard will just ignore extraneous values in the json that gets sent over the RPC. So it should be easy to test. But we should make dastard commander aware of these changes before merging.

### CODE REVIEW

File: lancero_source.go [hunk 1]
```diff
@@ -368,6 +370,46 @@ func roundint(x float64) int {
 	return int(x + math.Copysign(0.5, x))
 }
 
+type cringeGlobals struct {
+	SETT             int `json:"SETT"`
```
Comment (joefowler) [inline lancero_source.go]: Change Golang variable to `Settle` or `SETTLE`.

File: lancero_source.go [hunk 2]
```diff
@@ -368,6 +370,46 @@ func roundint(x float64) int {
...
+	SquenceLength    int `json:"seqln"`
```
Comment (joefowler) [inline lancero_source.go]: Don't you mean `SequenceLength` (properly spelled)?

File: lancero_source.go [hunk 3]
```diff
@@ -368,6 +370,46 @@ func roundint(x float64) int {
...
+	// jsonFile's content into 'cg'
+	err = json.Unmarshal(byteValue, &cg)
+	if err != nil {
+		return cringeGlobals{}, err
+	}
+	return cg, nil
+}
+
+// InjestCringeGlobals changes NSAMP, nrows and lsync to match the values from cg
+func (ls *LanceroSource) InjestCringeGlobals(cg cringeGlobals) {
```
Comment (joefowler) [inline lancero_source.go]: Spell `Ingest`. I'm not saying this in jest.

**Comment by ggggggggg** [comment]
Oddly I think you commented on an earlier commit. When I went to make your changes `SETT` was named `Sett`, the `Injest...` method had been removed and `SequenceLength` was spelled correctly. Did you click on the "File Changed" tab?

**Comment by joefowler** [comment]
I get confused in PR reviews. I must have reviewed the first commit, not the overall change. Hmm. Well, whatever. Looks fine to me.

**Pull request merged**